numpy/testing/_private/utils.py

```diff
@@ -784,18 +785,18 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
 
         if isinstance(val, bool):
             cond = val
-            reduced = [0]
+            reduced = array([0])
         else:
             reduced = val.ravel()
             cond = reduced.all()
-            reduced = reduced.tolist()
 
         # The below comparison is a hack to ensure that fully masked
         # results, for which val.ravel().all() returns np.ma.masked,
         # do not trigger a failure (np.ma.masked != True evaluates as
         # np.ma.masked, which is falsy).
         if cond != True:
-            mismatch = 100.0 * reduced.count(0) / ox.size
+            logical_not(reduced, out=reduced)
+            mismatch = 100.0 * reduced.sum(dtype=intp) / ox.size
```

Comment:
nit: faster to sum then subtract from 100:
```python
match = 100 * reduced.sum(dtype=intp) / ox.size
mismatch = 100 - match
```

Reply:
Or heck, subtract from `1` and use `%` in the format string

Reply:
The issue came about when testing with arrays that included `nan` or `inf` and the edge cases associated with those. 

I think the correct formula is
```
mismatch = 100. * (reduced.size - reduced.sum(dtype=intp)) / ox.size
```

Reply:
Its not so easy to get the same format
```
$ print(f'{1.113:.3g}')
1.11
$ print(f'{.113:.3%}')
11.300%
$ print(f'{.113:3%}')
11.300000%
```

The previous output would give 3 significant digits (well depends on how you count the leading `1` as a significant digit, but here I am counting it). So you would get `100%`, but `10.1` and `1.01`, but with the `%` it isn't obvious to me how to acheive that.

Reply:
That's awkward, fine to leave as is then